File: Linear/Dynamic/Queue/Queue.py
from collections import deque

class Queue:
    def __init__(self):
        self.q = deque()
    
    def enqueue(self, value):
        self.q.append(value)
    
    def dequeue(self):
        if self.q:
            self.q.popleft()
        else:
            print("Queue is empty")
    
    def getFront(self):
        if self.q:
            return self.q[0]
        else:
            print("Queue is empty")
    
    def getRear(self):
        if self.q:
            return self.q[-1]
        else:
            print("Queue is empty")
    
    def isEmpty(self):
        return len(self.q) == 0
    
    def size(self):
        return len(self.q)

q = Queue()

q.enqueue(1)
q.enqueue(8)
q.enqueue(3)

print("Front:", q.getFront())

print("Rear:", q.getRear())

q.dequeue()
print("After dequeue, front:", q.getFront())

print("Size:", q.size())

print("Is empty:", "Yes" if q.isEmpty() else "No")

File: Linear/Dynamic/Queue/Queue.js
class Queue {
    constructor() {
        this.q = []
    }

    enqueue(value) {
        this.q.push(value)
    }

    dequeue() {
        if (this.q.length > 0) {
            this.q.shift()
        } else {
            console.log("Queue is empty")
        }
    }

    getFront() {
        if (this.q.length > 0) {
            return this.q[0]
        } else {
            console.log("Queue is empty")
        }
        return null
    }

    getRear() {
        if (this.q.length > 0) {
            return this.q[this.q.length - 1]
        } else {
            console.log("Queue is empty")
        }
        return null
    }

    isEmpty() {
        return this.q.length === 0
    }

    size() {
        return this.q.length
    }
}

const q = new Queue()

q.enqueue(1)
q.enqueue(8)
q.enqueue(3)

console.log("Front:", q.getFront())

console.log("Rear:", q.getRear())

q.dequeue()
console.log("After dequeue, front:", q.getFront())

console.log("Size:", q.size())

console.log("Is empty:", q.isEmpty() ? "Yes" : "No")